File: php/src/Model/Attachment.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * An attachment record; `status` is `pending` until the upload is confirmed.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class Attachment
{
    public function __construct(
        public readonly string $id,
        public readonly string $taskId,
        public readonly string $filename,
        public readonly string $contentType,
        public readonly int $sizeBytes,
        public readonly string $status,
        public readonly AttachmentUploader $uploader,
        public readonly int $createdAt,
        public readonly ?int $confirmedAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            taskId: (string) ($data['taskId'] ?? ''),
            filename: (string) ($data['filename'] ?? ''),
            contentType: (string) ($data['contentType'] ?? ''),
            sizeBytes: (int) ($data['sizeBytes'] ?? 0),
            status: (string) ($data['status'] ?? ''),
            uploader: AttachmentUploader::fromArray($data['uploader'] ?? []),
            createdAt: (int) ($data['createdAt'] ?? 0),
            confirmedAt: isset($data['confirmedAt']) ? (int) $data['confirmedAt'] : null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/Attachment.java
package io.fivexer.sdk.model;

/**
 * An attachment record. {@code status} is {@code pending} until the upload is confirmed.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class Attachment {
    private String id;
    private String taskId;
    private String filename;
    private String contentType;
    private long sizeBytes;
    private String status;
    private AttachmentUploader uploader;
    private long createdAt;
    private Long confirmedAt;

    public String getId() { return id; }
    public String getTaskId() { return taskId; }
    public String getFilename() { return filename; }
    public String getContentType() { return contentType; }
    public long getSizeBytes() { return sizeBytes; }
    public String getStatus() { return status; }
    public AttachmentUploader getUploader() { return uploader; }
    public long getCreatedAt() { return createdAt; }
    public Long getConfirmedAt() { return confirmedAt; }
}
